beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestIntegrationInvoiceWithRepairLogic.java

```diff
@@ -890,8 +823,137 @@ public void testWithSuperflousRepairedItems() throws Exception {
                 new ExpectedInvoiceItemCheck(new LocalDate(2012, 6, 1), new LocalDate(2012, 7, 1), InvoiceItemType.RECURRING, new BigDecimal("249.95")));
         invoiceChecker.checkInvoice(invoices.get(2).getId(), callContext, toBeChecked);
 
+    }
+
+    //
+    //    Full item adjustments scenarios
+    //
+    //    Case V1: Cancellation EOT => Balance is $0 and nothing to regenerate
+    //    Case V2: Cancellation IMM => Balance is $0 and regenerate the new piece (proposed item)
+    //    Case V3: Cancellation EOT => Balance is $0 and nothing to regenerate  **** Our weird behavior ov V3 behaving like V1 ****
+
+
+    @Test(groups = "slow")
+    public void testRepairWithFullItemAdjustmentV1() throws Exception {
+
+
+        DefaultEntitlement bpEntitlement = setupTestRepairWithFullItemAdjustment();
+
+        // Move clock to 2013-09-17
+        clock.addDays(30);
+        busHandler.pushExpectedEvents(NextEvent.BLOCK, NextEvent.CANCEL, NextEvent.NULL_INVOICE);
+        bpEntitlement.cancelEntitlementWithPolicyOverrideBillingPolicy(EntitlementActionPolicy.IMMEDIATE, BillingActionPolicy.START_OF_TERM, ImmutableList.<PluginProperty>of(), callContext);
+        assertListenerStatus();
+
+        List<Invoice>  invoices = invoiceUserApi.getInvoicesByAccount(bpEntitlement.getAccountId(), false, false, callContext);
+        assertEquals(invoices.size(), 2);
+        ImmutableList<ExpectedInvoiceItemCheck> toBeChecked = ImmutableList.<ExpectedInvoiceItemCheck>of(
```

Comment:
Nit: could do the check before moving the clock to show that these items already existed and nothing was generated.

Reply:
@pierre  Isn't it what [this line](https://github.com/killbill/killbill/blob/work-for-invoice-0.21.x/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestIntegrationInvoiceWithRepairLogic.java#L954) is doing though?